File: 3251-3500/3300. Minimum Element After Replacement With Digit Sum/solution.cpp
#include <bits/stdc++.h> 

using namespace std ; 

class Solution {
public:
    int getSum( int n ){
        int ans = 0;
        while( n >0 ){
            ans += n % 10 ; 
            n /= 10 ; 
        } 
        return ans ; 
    }
    int minElement(vector<int>& nums) {
        int ans = INT_MAX; 
        for( int num : nums ){
            ans = min( ans , getSum(num) ) ;  
        }
        return ans ; 
    }
};

File: 3251-3500/3300. Minimum Element After Replacement With Digit Sum/solution.rs
impl Solution {
    pub fn min_element(nums: Vec<i32>) -> i32 {
      let n = nums.len() ; 
      let mut ans = 1_000_000_000 ; 
      for i in 0..n {
        let mut curr = nums[i] ; 
        let mut can = 0 ; 
        while curr > 0 { 
            can += curr % 10 ; 
            curr /= 10 ; 
        }
        ans = ans.min(can);
      } 
      return ans ;  
    }
}